src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2789,78 +2805,63 @@ export class ResultsViewPageStore {
         },
     });
 
-    readonly mutationMapperStores = remoteData<{
-        [hugoGeneSymbol: string]: ResultsViewMutationMapperStore;
-    }>(
-        {
-            await: () => [
-                this.genes,
+    @computed get selectedGene() {
+        return this.genes.result
+            ? _.find(
+                  this.genes.result,
+                  gene => gene.hugoGeneSymbol === this.selectedGeneSymbol
+              )
+            : undefined;
+    }
+
+    @computed get mutationMapperStoreForSelectedGene() {
+        if (
+            this.genes.isComplete &&
+            this.oncoKbCancerGenes.isComplete &&
+            this.oncoKbCancerGenes &&
+            this.uniqueSampleKeyToTumorType &&
+            this.mutations &&
+            this.mutationsByGene &&
+            this.selectedGene
+        ) {
+            return new ResultsViewMutationMapperStore(
```

Comment:
@onursumer yeah it's not that ideal, the purpose of this is to help not overwhelm genome nexus, so it can only send one gene instead of all genes, do you have better suggestions to do this?

Reply:
I think we can handle this by lazy initializing `MutationMapperStore`s in the `getMutationMapperStore(hugoSymbol: string)` function. That would eliminate the need to recompute the store upon a new gene selection action. We can define a cache of `[hugoGeneSymbol: string]: ResultsViewMutationMapperStore` in this store to make sure that each store is initialized only once.

Reply:
thanks!